### PR TITLE
RadialBar refactor and types

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Preparing for selectors.

## Related Issue

https://github.com/recharts/recharts/issues/4583

### CODE REVIEW

File: src/polar/RadialBar.tsx [hunk 1]
```diff
@@ -416,83 +536,27 @@ export class RadialBar extends PureComponent<RadialBarProps> {
...
-        value: stackedData ? value : value[1],
-        cx,
-        cy,
-        innerRadius,
-        outerRadius,
-        startAngle,
-        endAngle,
-        ...(cells && cells[index] && cells[index].props),
-        // @ts-expect-error missing types
-        tooltipPayload: [getTooltipItem(item, entry)],
```
Comment (PavelVanecek) [inline src/polar/RadialBar.tsx:493]: Hmmmm I did not include the tooltip payload and position. Do we have RadialBar tooltip coming from redux already?

Comment (ckifer) [reply src/polar/RadialBar.tsx:493]: I think yes? lol

Tooltip is still defaulted to context and falls back on redux though

Comment (PavelVanecek) [reply src/polar/RadialBar.tsx:493]: Works fine in my storybook

image

Comment (ckifer) [reply src/polar/RadialBar.tsx:493]: I don't think it was ever dependent on this particular line. This was just so the tooltip data for each item is included in each item

Comment (PavelVanecek) [reply src/polar/RadialBar.tsx:493]: Visual diff https://www.chromatic.com/build?appId=63da8268a0da9970db6992aa&number=1518

Comment (ckifer) [reply src/polar/RadialBar.tsx:493]: Yep lgtm

**Review by ckifer — APPROVED** [review]

**Pull request merged**